### ISSUE TITLE
Using ggeffects with nestedLogit models: New vignette

### ISSUE THREAD

**Issue by friendly** [strengejacke/ggeffects#671]
I appreciate the support for `nestedLogit` models in the `ggeffects` package.

You don't illustrate them here, so I tried to do this in a new pkgdown article, https://friendly.github.io/nestedLogit/articles/ggeffects.html 

I ran into some limitations in what I could do, but perhaps I'm missing something. Could you take a look at this vignette?

**Comment by strengejacke** [comment]
I get a 404 error?

<img width="396" height="217" alt="Image" src="https://github.com/user-attachments/assets/9b97e89f-75a9-4f42-a52b-680fbe61b997" />

**Comment by strengejacke** [comment]
I read the vignette from the source-files. I see the limitation now.
Usually, you can pass further arguments to `predict()`. However, the argument `model`, which is used to specify the sub-models, is already used in `predict_response()`. Maybe you can add an alias, like `submodel`, which may resolve this issue?

Or in *ggeffects*, we can have such an argument named `submodel`, that is passed to `predict(model = submodel)`.

**Comment by friendly** [comment]
I'm not sure what you mean. A `nestedLogit` object contains the $m-1$ submodels for the dichotomies in the `models` component.
These can be extracted using `nestedLogit::models()`

```
> names(wlf.nested)
[1] "models"          "formula"         "dichotomies"     "data"            "data.name"       "subset"         
[7] "contrasts"       "contrasts.print"

 # extract the binomial logit model for working vs. non-working
  m_work <- models(wlf.nested, "work")
```
Where would I add an alias?

Also, I'm confused about `predict_response()`, `ggpredict()` and `get_predictions_nestedLogit()`.  I can get something similar to the plot from `predict_response()` via

```
> pred <- ggpredict(wlf.nested, terms = c("hincome [all]", "children") )
> plot(pred)
```

The limitation I'm referring to is:

>The `ggeffects` package computes and plots predicted probabilities for the
_response categories_ of a nested logit model. As shown above, these can be
displayed on the logit scale using `ggplot2`'s built-in axis transformation.

>However, `ggeffects` does not currently provide access to the individual
dichotomy sub-models that comprise the nested logit --- for example, plotting
predicted values for the `work` and `full` dichotomies separately.

Note that the `predict.nestedLogit()` method does the right thing for the cases I want to illustrate. Maybe I or you could use this to show features of `ggeffects` better. I'm not showing here the use of `newdata` to get predictions over a grid for plotting.

```
> wlf.pred.nested <- predict(wlf.nested, model = "nested")
> str(wlf.pred.nested)
List of 4
 $ p       :'data.frame':	263 obs. of  3 variables:
  ..$ not.work: num [1:263] 0.706 0.688 0.895 0.771 0.74 ...
  ..$ parttime: num [1:263] 0.202 0.199 0.103 0.192 0.201 ...
  ..$ fulltime: num [1:263] 0.09234 0.11291 0.00189 0.03721 0.05974 ...
 $ logit   :'data.frame':	263 obs. of  3 variables:
  ..$ not.work: num [1:263] 0.874 0.79 2.144 1.213 1.044 ...
  ..$ parttime: num [1:263] -1.37 -1.39 -2.16 -1.44 -1.38 ...
  ..$ fulltime: num [1:263] -2.29 -2.06 -6.27 -3.25 -2.76 ...
 $ se.p    :'data.frame':	263 obs. of  3 variables:
  ..$ not.work: num [1:263] 0.0339 0.0351 0.0595 0.0424 0.0365 ...
  ..$ parttime: num [1:263] 0.0308 0.031 0.0585 0.0391 0.0337 ...
  ..$ fulltime: num [1:263] 0.02276 0.02493 0.00268 0.01765 0.02032 ...
 $ se.logit:'data.frame':	263 obs. of  3 variables:
  ..$ not.work: num [1:263] 0.163 0.164 0.633 0.24 0.19 ...
  ..$ parttime: num [1:263] 0.191 0.195 0.633 0.252 0.21 ...
  ..$ fulltime: num [1:263] 0.272 0.249 1.422 0.493 0.362 ...
 - attr(*, "class")= chr "predictNestedLogit"
> 
> wlf.pred.dichot <- predict(wlf.nested, model = "dichotomies")
> str(wlf.pred.dichot)
List of 2
 $ work:'data.frame':	263 obs. of  3 variables:
  ..$ fit           : num [1:263] -0.874 -0.79 -2.144 -1.213 -1.044 ...
  ..$ se.fit        : num [1:263] 0.163 0.164 0.633 0.24 0.19 ...
  ..$ residual.scale: num [1:263] 1 1 1 1 1 1 1 1 1 1 ...
 $ full:'data.frame':	263 obs. of  3 variables:
  ..$ fit           : num [1:263] -0.783 -0.568 -4.001 -1.641 -1.212 ...
  ..$ se.fit        : num [1:263] 0.317 0.298 1.325 0.521 0.402 ...
  ..$ residual.scale: num [1:263] 1 1 1 1 1 1 1 1 1 1 ...
 - attr(*, "model")= chr "wlf.nested"
 - attr(*, "dichotomies")= chr [1:2] "work" "full"
 - attr(*, "class")= chr "predictDichotomies"
 ```

**Comment by strengejacke** [comment]
> I'm not sure what you mean

The first argument in `ggeffects::predict_response()` and `modelbased::estimate_relation()` is named `model`. However, if I want to specify `model = "dichotomies"` in `predict.nestedLogit()`, I can't pass that argument down to `predict()` using, e.g., `ggeffects::predict_response(mymodel, model = "dichotomies")`, because then I have the argument `model` twice in my function call, which gives an error.

Thus, to fix the limitation you mentioned in your vignette, I would add an argument `submodel`, to avoid duplicated argument names.

**Comment by friendly** [comment]
Thanks for the follow-up & clarifying the relation.
That sounds sensible. Nested logit models do of course contain submodels.

**Comment by strengejacke** [comment]
Packages like `marginaleffects`, `ggeffects`, `modelbased` and also `insight::get_predicted()` have named their first argument `model`. You should probably think about renaming your `model`-argument in `predict()` into something else, to avoid the "duplicated argument" issue.

**Comment by friendly** [comment]
I understand the problem now, and I think for a future release of `nestedLogit` it would be worthwhile to change the argument `model` --> `type = c("nested", "dichotomies")`.

For now (v.0.4.0), I have a new vignette, https://friendly.github.io/nestedLogit/articles/ggeffects.html
which describes what can be done / or not using `ggeffects`. Take a look and let me know if there is anything unfair, particularly under **Limitations**

>The `ggeffects` package computes and plots predicted probabilities for the
_response categories_ of a nested logit model. As shown above, these can be
displayed on the logit scale using `ggplot2`'s built-in axis transformation.

>However, `ggeffects` does not currently provide access to the individual
dichotomy sub-models that comprise the nested logit --- for example, plotting
predicted values for the `work` and `full` dichotomies separately.
(This is due to a conflict in the names of arguments (`model`), and may be resolved
in a future version of `nestedLogit`.)

>For these more specialized displays, see `vignette("plotting-ggplot", package = "nestedLogit")`,
which describes a manual workflow using `predict()` with `model = "dichotomies"` and
`as.data.frame()` to construct fully customized `ggplot2` plots.

I'm thinking of submitting this version shortly.

**Comment by strengejacke** [comment]
Vignette / limitations sections looks good!

A few other comments:
- `type` is also a standard `predict()` argument (see `?predict.glm`), so maybe you could choose a different name for that argument?
- `insight::get_predicted()` and `modelbased::estimate_relation()` support dichotomies in their latest (GitHub) versions:

``` r
library(nestedLogit)
library(modelbased)

data(Womenlf, package = "carData")
comparisons <- logits(
  work = dichotomy("not.work", c("parttime", "fulltime")),
  full = dichotomy("parttime", "fulltime")
)

m <- nestedLogit(
  partic ~ hincome + children,
  dichotomies = comparisons,
  data = Womenlf
)

estimate_relation(m, by = "children")
#> Warning: Logistic regression model has a categorical response variable. You may
#>   need to set `include_response=TRUE` to make it work for predictions.
#> Model-based Predictions
#> 
#> children | Response | Predictions |   SE |       95% CI
#> -------------------------------------------------------
#> absent   | not.work | 0.33        | 0.05 | [0.23, 0.44]
#> absent   | fulltime | 0.58        | 0.06 | [0.47, 0.69]
#> absent   | parttime | 0.20        | 0.03 | [0.15, 0.27]
#> present  | parttime | 0.09        | 0.03 | [0.04, 0.18]
#> present  | not.work | 0.70        | 0.03 | [0.63, 0.77]
#> present  | fulltime | 0.09        | 0.02 | [0.06, 0.15]
#> 
#> Variable predicted: partic
#> Predictors modulated: children
#> Predictors controlled: hincome (15)
#> Predictions are on the response-scale.

estimate_relation(m, by = "children", submodel = "dichotomies")
#> Warning: Logistic regression model has a categorical response variable. You may
#>   need to set `include_response=TRUE` to make it work for predictions.
#> Model-based Predictions
#> 
#> children | Response | Predictions |   SE |       95% CI
#> -------------------------------------------------------
#> absent   |     work | 0.67        | 0.05 | [0.56, 0.77]
#> absent   |     full | 0.87        | 0.05 | [0.74, 0.94]
#> present  |     work | 0.30        | 0.03 | [0.23, 0.37]
#> present  |     full | 0.32        | 0.07 | [0.20, 0.46]
#> 
#> Variable predicted: partic
#> Predictors modulated: children
#> Predictors controlled: hincome (15)
#> Predictions are on the response-scale.

estimate_relation(m, by = "children", predict = "link")
#> Warning: Logistic regression model has a categorical response variable. You may
#>   need to set `include_response=TRUE` to make it work for predictions.
#> Model-based Predictions
#> 
#> children | Response | Predictions |   SE |         95% CI
#> ---------------------------------------------------------
#> absent   | not.work | -0.71       | 0.24 | [-1.19, -0.24]
#> absent   | fulltime | 0.34        | 0.23 | [-0.12,  0.79]
#> absent   | parttime | -1.38       | 0.19 | [-1.75, -1.00]
#> present  | parttime | -2.34       | 0.41 | [-3.15, -1.53]
#> present  | not.work | 0.86        | 0.16 | [ 0.54,  1.18]
#> present  | fulltime | -2.26       | 0.27 | [-2.78, -1.73]
#> 
#> Variable predicted: partic
#> Predictors modulated: children
#> Predictors controlled: hincome (15)
#> Predictions are on the link-scale.
```

<sup>Created on 2026-03-04 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by strengejacke** [comment]
I'll implement the `submodel` argument in ggeffects, too.

**Comment by strengejacke** [comment]
Ok, works with ggeffects now, too:

``` r
library(nestedLogit)
library(ggeffects)

data(Womenlf, package = "carData")
comparisons <- logits(
  work = dichotomy("not.work", c("parttime", "fulltime")),
  full = dichotomy("parttime", "fulltime")
)

m <- nestedLogit(
  partic ~ hincome + children,
  dichotomies = comparisons,
  data = Womenlf
)

predict_response(m, "children")
#> # Predicted probabilities of partic
#> 
#> partic: not.work
#> 
#> children | Predicted |     95% CI
#> ---------------------------------
#> absent   |      0.32 | 0.23, 0.43
#> present  |      0.70 | 0.63, 0.76
#> 
#> partic: parttime
#> 
#> children | Predicted |     95% CI
#> ---------------------------------
#> absent   |      0.08 | 0.04, 0.17
#> present  |      0.20 | 0.15, 0.27
#> 
#> partic: fulltime
#> 
#> children | Predicted |     95% CI
#> ---------------------------------
#> absent   |      0.60 | 0.48, 0.70
#> present  |      0.10 | 0.06, 0.16
#> 
#> Adjusted for:
#> * hincome = 14.00
predict_response(m, "children", submodel = "dichotomies")
#> # Predicted probabilities of partic
#> 
#> partic: work
#> 
#> children | Predicted |     95% CI
#> ---------------------------------
#> absent   |      0.68 | 0.57, 0.77
#> present  |      0.30 | 0.24, 0.37
#> 
#> partic: full
#> 
#> children | Predicted |     95% CI
#> ---------------------------------
#> absent   |      0.88 | 0.76, 0.94
#> present  |      0.34 | 0.22, 0.48
#> 
#> Adjusted for:
#> * hincome = 14.00
```

<sup>Created on 2026-03-04 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by friendly** [comment]
Great! Confirmed that this now works with `ggeffects` from the issue671 branch.

`modelbased::estimate_relation()` doesn't work for me, but perhaps this is a version mismatch. 

``` r
# install latest version
remotes::install_github("strengejacke/ggeffects", ref = "strengejacke/issue671")
#> Using GitHub PAT from the git credential store.
#> Skipping install of 'ggeffects' from a github remote, the SHA1 (dd7ffdc9) has not changed since last install.
#>   Use `force = TRUE` to force installation

library(nestedLogit)
library(modelbased)
library(ggeffects)
#> 
#> Attaching package: 'ggeffects'
#> The following objects are masked from 'package:modelbased':
#> 
#>     collapse_by_group, pool_predictions, residualize_over_grid

packageVersion("modelbased")
#> [1] '0.14.0.2'
# [1] ‘0.14.0.2’
packageVersion("ggeffects")
#> [1] '2.3.2.1'
# [1] ‘2.3.2.1’


data(Womenlf, package = "carData")
comparisons <- logits(
  work = dichotomy("not.work", c("parttime", "fulltime")),
  full = dichotomy("parttime", "fulltime")
)

m <- nestedLogit(
  partic ~ hincome + children,
  dichotomies = comparisons,
  data = Womenlf
)

estimate_relation(m, by = "children")
#> Warning: Logistic regression model has a categorical response variable. You may
#>   need to set `include_response=TRUE` to make it work for predictions.
#> Warning: Something went wrong with computing standard errors and confidence
#>   intervals for predictions.
#> Warning: Could not apply Delta method to transform standard errors.
#>   You may be able to obtain standard errors by using the 
#>   `predict="link"` argument value.
#> Error in `link_inv()`:
#> ! Argument eta must be a nonempty numeric vector
```

<sup>Created on 2026-03-04 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by strengejacke** [comment]
`modelbased::estimate_relation()` relies on `insight::get_predicted()`, thus you need to update some more packages. Easiest is to run `easystats::install_latest()`, then it should work.
